src/components/UserSelector.tsx

```diff
@@ -1,19 +1,63 @@
-import React from 'react';
+import React, {
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import classNames from 'classnames';
+import { User } from '../types/User';
+
+type Props = {
+  users: User[];
+  onUserClick: (userId: number) => void;
+};
+
+export const UserSelector: React.FC<Props> = React.memo(({
+  users,
+  onUserClick,
+}) => {
+  const [userName, setUserName] = useState('Choose a user');
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+  const dropdownRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const handleOutsideClick = (event: MouseEvent) => {
+      if (dropdownRef.current
+        && !dropdownRef.current.contains(event.target as Node)) {
+        setIsDropdownActive(false);
+      }
+    };
+
+    document.addEventListener('click', handleOutsideClick);
+
+    return () => {
+      document.removeEventListener('click', handleOutsideClick);
+    };
+  }, []);
+
+  const handleClick = useCallback((userId, name) => {
+    setUserName(name);
+    onUserClick(userId);
+    setIsDropdownActive(false);
+  }, [userName]);
```

Comment:
The dependency array of useCallback should include 'onUserClick' and 'userName' to keep the function up-to-date with the latest values.